src/commit.c

```diff
@@ -337,6 +338,15 @@ int git_commit__parse(void *_commit, git_odb_object *odb_obj)
 	if (git_signature__parse(commit->author, &buffer, buffer_end, "author ", '\n') < 0)
 		return -1;
 
+	/* Some tools create multiple author fields, ignore the extra ones */
+	while ((size_t)(buffer_end - buffer) >= strlen("author ") && !git__prefixcmp(buffer, "author ")) {
+		if (git_signature__parse(&dummy_sig, &buffer, buffer_end, "author ", '\n') < 0)
```

Comment:
We still expect it to be well-formed, otherwise it's just an invalid commit. We allow the multiple author fields because git lets them through and these commits exist in the wild, but if it just happens to have "author lalala\n" somewhere in the header, that doesn't mean we should ignore the fact that it's malformed.